connectors/db/vector.py

```diff
@@ -189,52 +260,99 @@ def init_vector_store(self):
         self.search_providers = [
             MMRSearchProvider(self.store),
             SimilaritySearchProvider(self.store),
+            HybridSearchProvider(self.store),
         ]
 
     def combine_small_chunks(self, chunks):
         """
-        Combine small chunks into the next chunk
-
-        Sometimes we see the text splitter create a chunk containing only a single line (like
-        a header), we will store these small chunks on the next chunk to avoid storing a
-        document with small context
+        Keep merging small chunks into the next one until the result is large enough.
         """
-        for idx, chunk in enumerate(chunks):
+        merged_chunks = []
+        buffer = ""
+
+        for chunk in chunks:
+            if len(chunk.strip()) == 0:  # Ignore empty chunks
+                continue
+
             if len(chunk) < 200:
-                # this chunk is less than 200 chars, move it to the next chunk
-                try:
-                    chunks[idx + 1] = f"{chunk}\n\n{chunks[idx + 1]}"
-                except IndexError:
-                    # we've reached the end and there is no 'next chunk', just give up
-                    break
-                # make note of its index and pop it later...
-                chunks[idx] = "<<removed>>"
+                buffer += chunk + "\n\n"  # Collect small chunks
+            else:
+                if buffer:
+                    chunk = buffer + chunk  # Merge collected buffer into this chunk
+                    buffer = ""  # Reset buffer
+                merged_chunks.append(chunk)
 
-        chunks = list(filter(lambda val: val != "<<removed>>", chunks))
+        # If anything is left in the buffer, add it to the last chunk
+        if buffer and merged_chunks:
+            merged_chunks[-1] += "\n\n" + buffer
 
-        return chunks
+        return merged_chunks
 
-    def split_to_document_chunks(self, text_to_split, metadata):
-        # find title if possible and add to metadata
-        for line in text_to_split.splitlines():
-            if line.startswith("# "):
-                # we found a title header, add it to metadata
-                metadata["title"] = line.lstrip("# ").strip()
-                break
-        else:
-            metadata["title"] = metadata["full_path"]
 
-        text_splitter = RecursiveCharacterTextSplitter(
-            chunk_size=self.vector_chunk_size,
-            chunk_overlap=self.vector_chunk_overlap,
-            separators=TXT_SEPARATORS,  # todo: split on more doc types
-        )
+    def has_markdown_headers(self, text):
+        """Checks if a document contains markdown headers."""
+        return bool(re.search(r"^#{1,6} ", text, re.MULTILINE))
+
+    def remove_tables_of_contents(self, chunks):
```

Comment:
I think we should remove this stuff *before* chunking? Somewhere in this function? https://github.com/RedHatInsights/tangerine-backend/blob/main/connectors/db/file.py#L173

Then you can re.search the whole doc at once.

Reply:
Good call.

Reply:
Added your code (thank you!) and removed mine